coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,243 @@
+
+Examples:
+```JS
+switch (a) {
+  case "a":
+    break;
+  case "b":
+    break;
+}
+
+var a,
+  b,
+  c;
+
+function test() {
+  var a,
+    b,
+    c;
+}
+
+test(
+  "A Very long string which is longer then one line, which is 80 characters " +
+    "in this case. This demonstrates how the second line is indented",
+  "Another string parameter.",
+  {and: "a Literal Object", as: "a parameter", type: "short"},
+  {
+    yet: "another long parameter Object",
+    which: "needs to be split in multiple",
+    lines: 3
+  }
+);
+
+test({
+  a: {
+    b: 10
+  },
+  c: 20,
+  d: 30,
+  e: 40,
+  f: 50,
+  g: 60,
+  h: {
+    i: 70
+  }
+});
+```
+
+## Whitespace
+Rules:
+- No Trailing spaces should be used at the end of a line.
+- No multiple Blank lines.
+
+## Functions
+Rules:
+  - Function blocks should have a space before the definition.
+  - Function should have a space before the opening parentheses.
+  - Named Functions should not have a space before the opening function parentheses.
+  - Functions can be on a single line.
+  - Anonymous Functions need to have a space before the opening function parentheses.
+  - Arrow functions do have the same constraints
+
+Examples:
+```JS
+function test() {
+  return;
+}
+
+function test() { var a = 10; return a; }
+
+var test = function () {var b = 10; return b;}
+
+var es6 = parameter => parameter();
+
+var es6 = parameter => {
+  parameter();
+  return false;
+}
+
+var es6 = (parameter, index) => parameter(index);
+
+var es6 = (parameter, index) => {
+  parameter(index);
+  return false;
+}
+```
+
+## Comments
+Rules:
+  - Comments should be prefixed with a space sign.
+  - The `*` sign is a exception to the rule.
+
+Examples:
+```JS
+// Single line Comment
+
+//* A Exception to the space rule
+
+/*
+  Comment Block with no space
+*/
+
+/**
+ * Block Comment
+ */
+```
+
+## If, loops, switch blocks
+Rules:
+  - Space after the keyword
+  - Space before opening parentheses
+  - Strict equal (`===` or `!==`)
+
+Examples:
+```JS
+if (a === b) {
+  a();
+}
+
+if (true !== false) {
+  // Statements
+}
+
+switch (a) {
+  case 1:
+    break;
+  case 2:
+    break;
+  default:
+    break;
+}
+
+for (var a = 10; a !== 0; a--) {
+  // Statements
+}
+```
+
+## Strings
+Strings are defined with double quotes.
+
+## Variables and Attribute names
+Variables always use camelCase names.
+May have a underscore dangle.
+Examples:
+```JS
+var camelCase = "{value}",
+  _camelCase = "{value}",
+  CamelCase = "{test}",
+  camelCase_ = "{test}";
+```
+
+## Object Literals
+Object literals should have no space after the '{' or before the '}' signs.
+
+Valid Example
+```JS
+var objectLiteral = {"key": "value", "key2": "value2"};
+
+var objectLiteral = {
+  "key": "value",
+  "key2": "value2"
+};
+```
+
+## JSDoc
+If there is a return there should be a type and description.
+If there is one or multiple parameter/s there should be a name, type and description for each one.
+
```

Comment:
I would add, that the function should be understandable without commenting, so it actually needs no comment header. A function should be descriptive by reading the code or the name.